src/njs_buffer.c

```diff
@@ -1618,6 +1618,8 @@ njs_buffer_write_string(njs_vm_t *vm, njs_value_t *value,
         goto done;
     }
 
+    length = njs_min(str.length, (size_t) length);
```

Comment:
This change does not cover edge case when there is no room for trailing multibyte character in a buffer,
        for example:
            var buf = Buffer.alloc(10);
            var len = buf.write('\u00bd', 0, 1);
        It should set len == 0, and write nothing to the buf.

        For referenece see quotation: "However, partially encoded characters will not be written."
        from https://nodejs.org/api/buffer.html#bufwritestring-offset-length-encoding